src/txn/txn.c

```diff
@@ -644,7 +644,12 @@ __wt_txn_release(WT_SESSION_IMPL *session)
 	/* Ensure the transaction flags are cleared on exit */
 	txn->flags = 0;
 	txn->prepare_timestamp = WT_TS_NONE;
-	txn->durable_timestamp = WT_TS_NONE;
+
+	/*
+	 * Purposely do NOT clear the commit and durable timestamps on release.
+	 * Other readers may still find these transactions in the durable queue
+	 * and will need to see those timestamps.
+	 */
```

Comment:
It's odd to have a comment not associated with any code block. I'd join this with the comment about clearing flags at line 644.

Reply:
3e0eab0

Moved the comment.